pitop/processing/algorithms/ball_detect.py

```diff
@@ -36,40 +36,60 @@ def __init__(self, color):
         }
         self.match_limit = match_limits[color]
 
-        self.center_points = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
-        self.radius = 0
-        self.angle_from_center = None
+        self._center_points_cv = deque(maxlen=DETECTION_POINTS_BUFFER_LENGTH)
+        self._center = None
+        self._radius = 0
+        self._angle = None
 
-    def clear(self):
-        self.center_points.clear()
+    def no_detection(self):
+        self.center_points_cv.appendleft(None)
+        self.center = None
         self.radius = 0
-        self.angle_from_center = None
+        self.angle = None
+
+    @property
+    def angle(self) -> float:
+        return self._angle
+
+    @angle.setter
+    def angle(self, value: float):
+        self._angle = value
 
     @property
-    def angle(self):
-        return self.angle_from_center
+    def center(self) -> tuple:
+        return self._center
+
+    @center.setter
+    def center(self, value: tuple):
+        self._center = value
+
+    @property
+    def radius(self) -> int:
+        return self._radius
+
+    @radius.setter
+    def radius(self, value: int):
+        self._radius = value
 
     @property
-    def center(self):
-        if len(self.center_points) > 0:
-            return self.center_points[0]
-        return None
+    def center_points_cv(self) -> deque:
+        return self._center_points_cv
 
-    def is_valid(self):
+    @property
+    def found(self) -> bool:
         return self.center is not None
 
 
 class BallDetector:
     def __init__(self,
-                 process_image_width: int = 320,
-                 format: str = "OpenCV",
-                 print_fps: bool = False):
+                 image_processing_width: int = 320,
```

Comment:
What if you want to specify its height? I think it would be better if you pass in a size tuple, with co-ordinate unset if you don't want to resize a particular axis. For example:
`image_processing_size=(320,)` would mean "resize to 320 width" as here, but is more flexible and more sanely named.

Reply:
This will be a useful paradigm to get right

Reply:
Well this would allow people to scale the width and height arbitrarily and we want it to be proportional. I see what you mean but I'd prefer to leave it as it is and just add a good description for the docs

Reply:
It would, yes, but that would be on them if they did that. If that is not in our content, and the docs say "use width OR height to keep proportions the same, which you probably want" then it would be fine, and we can reuse this anywhere we need to scale an image for processing as it's generic.